infra/src/main/kotlin/fi/fta/geoviite/infra/linking/SwitchLinkingService.kt

```diff
@@ -750,13 +761,21 @@ fun findClosestIntersections(
                     desiredLocation = desiredLocation
                 )
             } else {
-                val distance1 = pointDistanceToLine(line1.start, line1.end, line2.start)
-                val distance2 = pointDistanceToLine(line1.start, line1.end, line2.end)
-                val minDistance = min(distance1, distance2)
-                if (minDistance <= MAX_LINE_INTERSECTION_DISTANCE) {
+                val linePointDistanceCheckPairs = listOf(
+                    line1 to line2.start,
+                    line1 to line2.end,
+                    line2 to line1.start,
+                    line2 to line1.end
+                )
+                val minDistanceAndPoint = linePointDistanceCheckPairs
```

Comment:
Tuon parin voisi destructuroida jo tässä

Reply:
Katson tämän jos ymmärrän :)

Reply:
Eli meinasin ihan vain pair destructurointia joka menee kotlinissa näin:
```
val (point, distance) = linePointDistance....
```
-> ei tarvitse second/first kutsuja parista tuonne alle vaan saa suoraan erillisiin muuttujiin nuo sisällöt